syscalls.c

```diff
@@ -145,7 +150,8 @@ int syscalls_sys_mprotect(void *ustack)
 	GETFROMSTACK(ustack, size_t, len, 1);
 	GETFROMSTACK(ustack, int, prot, 2);
 
-	err = vm_mprotect(proc->mapp, vaddr, len, PROT_USER | prot);
+	/* parasoft-suppress-next-line MISRAC2012_RULE_10_3 "prot is popped from stack -> size of int stays" */
+	err = (int)vm_mprotect(proc->mapp, vaddr, len, PROT_USER | (unsigned int)prot);
```

Comment:
inconsistency, in `syscalls_sys_mmap` you've switched to `unsigned int` for variable definition and GETFROMSTACK macro.

be consistent.